data_handler/handlers/order_books/myswap/main.py

```diff
@@ -0,0 +1,240 @@
+            return
+        self.asks.extend(local_asks)
+
+    def add_bids(self, pool_bids: pd.DataFrame, price_range: tuple[Decimal, Decimal]) -> None:
+        """
+        Add asks data to the order book.
+        :param pool_bids: pd.DataFrame - Bids in the pool.
+        :param price_range: tuple[Decimal, Decimal] - The price range for filtering.
+        """
+        if pool_bids.empty:
+            return
+        local_bids = []
+        prev_tick = Decimal(pool_bids.iloc[0]['tick'].item())
+        prev_price = self.tick_to_price(prev_tick)
+        y = self._get_token_amount(
+            Decimal(pool_bids.iloc[0]['liq'].item()),
+            self.current_price.sqrt(),
+            prev_price.sqrt(),
+            is_ask=False
+        )
+        local_bids.append((prev_price, y))
+        for index, bid_info in enumerate(pool_bids.iloc[::-1].itertuples(index=False)):
+            if index == 0:
+                continue
+            current_tick = Decimal(pool_bids.iloc[index - 1]['tick'].item())
+            current_price = self.tick_to_price(current_tick)
+            y = self._get_token_amount(
+                Decimal(pool_bids.iloc[index]['liq'].item()),
+                current_price.sqrt(),
+                Decimal(self.tick_to_price(pool_bids.iloc[index]['tick'].item())).sqrt(),
+                is_ask=False
+            )
+            local_bids.append((current_price, y))
+        if self.apply_filtering:
+            self.bids.extend([bid for bid in local_bids if price_range[0] < bid[0] < price_range[1]])
+            return
+        self.bids.extend(local_bids)
+
+    def _get_token_amount(
+            self, current_liq: Decimal, current_sqrt: Decimal, next_sqrt: Decimal, is_ask: bool = True
+    ) -> Decimal:
+        """
+        Calculate token amount based on liquidity data and current data processed(asks/bids).
+        :param current_liq: Decimal - Current price liquidity
+        :param current_sqrt: Decimal - Current square root of a price
+        :param next_sqrt: Decimal - Next square root of a price
+        :param is_ask: bool - True if an ask data
+        :return: Decimal - token amount
+        """
+        if is_ask and (current_sqrt == 0 or next_sqrt == 0):
+            raise ValueError("Square root of prices for asks can't be zero.")
+        if not is_ask and next_sqrt == 0:
+            return abs(current_liq / current_sqrt) / self._decimals_diff
+        amount = abs(current_liq / next_sqrt - current_liq / current_sqrt)
+        return amount / self._decimals_diff
+
+    def tick_to_price(self, tick: Decimal) -> Decimal:
+        return Decimal("1.0001") ** (tick - MAX_MYSWAP_TICK) * Decimal(2 ** 128) * self._decimals_diff
+
+    def calculate_liquidity_amount(self, tick, liquidity_pair_total) -> Decimal:
+        sqrt_ratio = self.get_sqrt_ratio(tick)
+        liquidity_delta = liquidity_pair_total / (sqrt_ratio / Decimal(2 ** 128))
+        return liquidity_delta / 10 ** self.token_a_decimal
```

Comment:
DITTO (add reference/comment on the formula)

Reply:
This method requires implementation in abstract class, but isn't used anywhere. I copied formula from existing method in Ekubo.